test/GeneratorTests.cpp

```diff
@@ -23,3 +16,21 @@ TEST_F(GeneratorTests, checkConversionFromIntToFieldsEnum)
 
     ASSERT_EQ(result, ConvertIntToFieldEnum(input));
 }
+
+TEST_F(GeneratorTests, checkSizeOfSAllPermutationOfBoard2x3)
+{
+    int rows = 2;
+    int cols = 3;
+    auto result = GenerateAllBoardPermutations(rows,cols);
+    std::set<Lines> s(result.begin(), result.end());
+    ASSERT_EQ(s.size(), pow(2,(rows * cols)));
+}
+
+TEST_F(GeneratorTests, checkSizeOfSAllPermutationOfBoard4x3)
+{
+    int rows = 4;
+    int cols = 3;
+    auto result = GenerateAllBoardPermutations(rows,cols);
+    std::set<Lines> s(result.begin(), result.end());
+    ASSERT_EQ(s.size(), pow(2,(rows * cols)));
```

Comment:
It's the same test as a previous one, only values for rows and cols differ. You can use TEST_P to write test case only once and check it on different inputs

Reply:
Added parameterized test case.